# FIFA

## Importing Required Modules

1. Importing numpy for mathematical operation on arrays and dataframes.
2. Importing pandas for reading data and data manipulation.

In [66]:
import numpy as np
import pandas as pd

## Reading Dataset

In [93]:
fifa=pd.read_csv("C:\\Users\\ritik\\Downloads\\datasets-20221227T153602Z-001\\datasets\\data sets\\fifa_data.csv")
fifa

Unnamed: 0      ID                Name  Age  \
0               0  158023            L. Messi   31   
1               1   20801   Cristiano Ronaldo   33   
2               2  190871           Neymar Jr   26   
3               3  193080              De Gea   27   
4               4  192985        K. De Bruyne   27   
...           ...     ...                 ...  ...   
18202       18202  238813        J. Lundstram   19   
18203       18203  243165  N. Christoffersson   19   
18204       18204  241638           B. Worman   16   
18205       18205  246268      D. Walker-Rice   17   
18206       18206  246269           G. Nugent   16   

                                                Photo Nationality  \
0      https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1       https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2      https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3      https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4      https://cdn.sofifa.org/players/4/19/192985.png     Belgium   
...                                               ...         ...   
18202  https://cdn.sofifa.org/players/4/19/238813.png     England   
18203  https://cdn.sofifa.org/players/4/19/243165.png      Sweden   
18204  https://cdn.sofifa.org/players/4/19/241638.png     England   
18205  https://cdn.sofifa.org/players/4/19/246268.png     England   
18206  https://cdn.sofifa.org/players/4/19/246269.png     England   

                                      Flag  Overall  Potential  \
0      https://cdn.sofifa.org/flags/52.png       94         94   
1      https://cdn.sofifa.org/flags/38.png       94         94   
2      https://cdn.sofifa.org/flags/54.png       92         93   
3      https://cdn.sofifa.org/flags/45.png       91         93   
4       https://cdn.sofifa.org/flags/7.png       91         92   
...                                    ...      ...        ...   
18202  https://cdn.sofifa.org/flags/14.png       47         65   
18203  https://cdn.sofifa.org/flags/46.png       47         63   
18204  https://cdn.sofifa.org/flags/14.png       47         67   
18205  https://cdn.sofifa.org/flags/14.png       47         66   
18206  https://cdn.sofifa.org/flags/14.png       46         66   

                      Club  ... Composure Marking StandingTackle  \
0             FC Barcelona  ...      96.0    33.0           28.0   
1                 Juventus  ...      95.0    28.0           31.0   
2      Paris Saint-Germain  ...      94.0    27.0           24.0   
3        Manchester United  ...      68.0    15.0           21.0   
4          Manchester City  ...      88.0    68.0           58.0   
...                    ...  ...       ...     ...            ...   
18202      Crewe Alexandra  ...      45.0    40.0           48.0   
18203       Trelleborgs FF  ...      42.0    22.0           15.0   
18204     Cambridge United  ...      41.0    32.0           13.0   
18205      Tranmere Rovers  ...      46.0    20.0           25.0   
18206      Tranmere Rovers  ...      43.0    40.0           43.0   

       SlidingTackle GKDiving  GKHandling  GKKicking  GKPositioning  \
0               26.0      6.0        11.0       15.0           14.0   
1               23.0      7.0        11.0       15.0           14.0   
2               33.0      9.0         9.0       15.0           15.0   
3               13.0     90.0        85.0       87.0           88.0   
4               51.0     15.0        13.0        5.0           10.0   
...              ...      ...         ...        ...            ...   
18202           47.0     10.0        13.0        7.0            8.0   
18203           19.0     10.0         9.0        9.0            5.0   
18204           11.0      6.0         5.0       10.0            6.0   
18205           27.0     14.0         6.0       14.0            8.0   
18206           50.0     10.0        15.0        9.0           12.0   

      GKReflexes Release Clause  
0            8.0        €226.5M 

## To read 10 rows from starting`

In [68]:
fifa.head(10)

Unnamed: 0      ID               Name  Age  \
0           0  158023           L. Messi   31   
1           1   20801  Cristiano Ronaldo   33   
2           2  190871          Neymar Jr   26   
3           3  193080             De Gea   27   
4           4  192985       K. De Bruyne   27   
5           5  183277          E. Hazard   27   
6           6  177003          L. Modrić   32   
7           7  176580          L. Suárez   31   
8           8  155862       Sergio Ramos   32   
9           9  200389           J. Oblak   25   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   
5  https://cdn.sofifa.org/players/4/19/183277.png     Belgium   
6  https://cdn.sofifa.org/players/4/19/177003.png     Croatia   
7  https://cdn.sofifa.org/players/4/19/176580.png     Uruguay   
8  https://cdn.sofifa.org/players/4/19/155862.png       Spain   
9  https://cdn.sofifa.org/players/4/19/200389.png    Slovenia   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   
5   https://cdn.sofifa.org/flags/7.png       91         91   
6  https://cdn.sofifa.org/flags/10.png       91         91   
7  https://cdn.sofifa.org/flags/60.png       91         91   
8  https://cdn.sofifa.org/flags/45.png       91         91   
9  https://cdn.sofifa.org/flags/44.png       90         93   

                  Club  ... Composure Marking StandingTackle  SlidingTackle  \
0         FC Barcelona  ...      96.0    33.0           28.0           26.0   
1             Juventus  ...      95.0    28.0           31.0           23.0   
2  Paris Saint-Germain  ...      94.0    27.0           24.0           33.0   
3    Manchester United  ...      68.0    15.0           21.0           13.0   
4      Manchester City  ...      88.0    68.0           58.0           51.0   
5              Chelsea  ...      91.0    34.0           27.0           22.0   
6          Real Madrid  ...      84.0    60.0           76.0           73.0   
7         FC Barcelona  ...      85.0    62.0           45.0           38.0   
8          Real Madrid  ...      82.0    87.0           92.0           91.0   
9      Atlético Madrid  ...      70.0    27.0           12.0           18.0   

  GKDiving  GKHandling  GKKicking  GKPositioning GKReflexes Release Clause  
0      6.0        11.0       15.0           14.0        8.0        €226.5M  
1      7.0        11.0       15.0           14.0       11.0        €127.1M  
2      9.0         9.0       15.0           15.0       11.0        €228.1M  
3     90.0        85.0       87.0           88.0       94.0        €138.6M  
4     15.0        13.0        5.0           10.0       13.0        €196.4M  
5     11.0        12.0        6.0            8.0        8.0        €172.1M  
6     13.0         9.0        7.0           14.0        9.0        €137.4M  
7     27.0        25.0       31.0           33.0       37.0          €164M  
8     11.0         8.0        9.0            7.0       11.0        €104.6M  
9     86.0        92.0       78.0           88.0       89.0        €144.5M  

[10 rows x 89 columns]

## To read 5 rows from ending

In [69]:
fifa.tail()

Unnamed: 0      ID                Name  Age  \
18202       18202  238813        J. Lundstram   19   
18203       18203  243165  N. Christoffersson   19   
18204       18204  241638           B. Worman   16   
18205       18205  246268      D. Walker-Rice   17   
18206       18206  246269           G. Nugent   16   

                                                Photo Nationality  \
18202  https://cdn.sofifa.org/players/4/19/238813.png     England   
18203  https://cdn.sofifa.org/players/4/19/243165.png      Sweden   
18204  https://cdn.sofifa.org/players/4/19/241638.png     England   
18205  https://cdn.sofifa.org/players/4/19/246268.png     England   
18206  https://cdn.sofifa.org/players/4/19/246269.png     England   

                                      Flag  Overall  Potential  \
18202  https://cdn.sofifa.org/flags/14.png       47         65   
18203  https://cdn.sofifa.org/flags/46.png       47         63   
18204  https://cdn.sofifa.org/flags/14.png       47         67   
18205  https://cdn.sofifa.org/flags/14.png       47         66   
18206  https://cdn.sofifa.org/flags/14.png       46         66   

                   Club  ... Composure Marking StandingTackle  SlidingTackle  \
18202   Crewe Alexandra  ...      45.0    40.0           48.0           47.0   
18203    Trelleborgs FF  ...      42.0    22.0           15.0           19.0   
18204  Cambridge United  ...      41.0    32.0           13.0           11.0   
18205   Tranmere Rovers  ...      46.0    20.0           25.0           27.0   
18206   Tranmere Rovers  ...      43.0    40.0           43.0           50.0   

      GKDiving  GKHandling  GKKicking  GKPositioning GKReflexes Release Clause  
18202     10.0        13.0        7.0            8.0        9.0          €143K  
18203     10.0         9.0        9.0            5.0       12.0          €113K  
18204      6.0         5.0       10.0            6.0       13.0          €165K  
18205     14.0         6.0       14.0            8.0        9.0          €143K  
18206     10.0        15.0        9.0           12.0        9.0          €165K  

[5 rows x 89 columns]

## To check the rows and columns in the data 

In [70]:
fifa.shape

(18207, 89)

## To check all columns in the data

In [71]:
fifa.columns

Index(['Unnamed: 0', 'ID', 'Name', 'Age', 'Photo', 'Nationality', 'Flag',
       'Overall', 'Potential', 'Club', 'Club Logo', 'Value', 'Wage', 'Special',
       'Preferred Foot', 'International Reputation', 'Weak Foot',
       'Skill Moves', 'Work Rate', 'Body Type', 'Real Face', 'Position',
       'Jersey Number', 'Joined', 'Loaned From', 'Contract Valid Until',
       'Height', 'Weight', 'LS', 'ST', 'RS', 'LW', 'LF', 'CF', 'RF', 'RW',
       'LAM', 'CAM', 'RAM', 'LM', 'LCM', 'CM', 'RCM', 'RM', 'LWB', 'LDM',
       'CDM', 'RDM', 'RWB', 'LB', 'LCB', 'CB', 'RCB', 'RB', 'Crossing',
       'Finishing', 'HeadingAccuracy', 'ShortPassing', 'Volleys', 'Dribbling',
       'Curve', 'FKAccuracy', 'LongPassing', 'BallControl', 'Acceleration',
       'SprintSpeed', 'Agility', 'Reactions', 'Balance', 'ShotPower',
       'Jumping', 'Stamina', 'Strength', 'LongShots', 'Aggression',
       'Interceptions', 'Positioning', 'Vision', 'Penalties', 'Composure',
       'Marking', 'StandingTackle', 'SlidingT

## To check the datatypes of the dataset

In [72]:
fifa.dtypes[1:20]

ID                            int64
Name                         object
Age                           int64
Photo                        object
Nationality                  object
Flag                         object
Overall                       int64
Potential                     int64
Club                         object
Club Logo                    object
Value                        object
Wage                         object
Special                       int64
Preferred Foot               object
International Reputation    float64
Weak Foot                   float64
Skill Moves                 float64
Work Rate                    object
Body Type                    object
dtype: object

In [73]:
fifa.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18207 entries, 0 to 18206
Data columns (total 89 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                18207 non-null  int64  
 1   ID                        18207 non-null  int64  
 2   Name                      18207 non-null  object 
 3   Age                       18207 non-null  int64  
 4   Photo                     18207 non-null  object 
 5   Nationality               18207 non-null  object 
 6   Flag                      18207 non-null  object 
 7   Overall                   18207 non-null  int64  
 8   Potential                 18207 non-null  int64  
 9   Club                      17966 non-null  object 
 10  Club Logo                 18207 non-null  object 
 11  Value                     18207 non-null  object 
 12  Wage                      18207 non-null  object 
 13  Special                   18207 non-null  int64  
 14  Prefer

## Height and weight should be in float 


so changing the height and weight type 

In [74]:
fifa["Height"].unique()

array(["5'7", "6'2", "5'9", "6'4", "5'11", "5'8", "6'0", "5'6", "5'10",
       "6'6", "6'1", "5'4", "6'3", "5'5", "6'5", "6'7", "5'3", "5'2",
       "6'8", "5'1", "6'9", nan], dtype=object)

In [75]:
fifa["Height"]=fifa["Height"].str.replace("'",".")
fifa["Height"].unique()

array(['5.7', '6.2', '5.9', '6.4', '5.11', '5.8', '6.0', '5.6', '5.10',
       '6.6', '6.1', '5.4', '6.3', '5.5', '6.5', '6.7', '5.3', '5.2',
       '6.8', '5.1', '6.9', nan], dtype=object)

In [76]:
fifa["Height"]=fifa["Height"].astype(float)
fifa["Height"]

0        5.70
1        6.20
2        5.90
3        6.40
4        5.11
         ... 
18202    5.90
18203    6.30
18204    5.80
18205    5.10
18206    5.10
Name: Height, Length: 18207, dtype: float64

In [77]:
fifa["Weight"].unique()

array(['159lbs', '183lbs', '150lbs', '168lbs', '154lbs', '163lbs',
       '146lbs', '190lbs', '181lbs', '192lbs', '176lbs', '172lbs',
       '148lbs', '165lbs', '196lbs', '161lbs', '187lbs', '212lbs',
       '170lbs', '203lbs', '157lbs', '185lbs', '130lbs', '174lbs',
       '207lbs', '134lbs', '141lbs', '152lbs', '179lbs', '132lbs',
       '201lbs', '198lbs', '209lbs', '214lbs', '143lbs', '205lbs',
       '137lbs', '194lbs', '216lbs', '139lbs', '220lbs', '126lbs',
       '218lbs', '123lbs', '227lbs', '128lbs', '223lbs', '225lbs',
       '121lbs', '115lbs', '117lbs', '236lbs', '229lbs', '243lbs', nan,
       '110lbs', '119lbs', '234lbs'], dtype=object)

In [78]:
fifa["Weight"]=fifa["Weight"].str.replace("lbs","")
fifa["Weight"].unique()

array(['159', '183', '150', '168', '154', '163', '146', '190', '181',
       '192', '176', '172', '148', '165', '196', '161', '187', '212',
       '170', '203', '157', '185', '130', '174', '207', '134', '141',
       '152', '179', '132', '201', '198', '209', '214', '143', '205',
       '137', '194', '216', '139', '220', '126', '218', '123', '227',
       '128', '223', '225', '121', '115', '117', '236', '229', '243', nan,
       '110', '119', '234'], dtype=object)

In [79]:
fifa["Weight"]=fifa["Weight"].astype(float)
fifa["Weight"].unique()

array([159., 183., 150., 168., 154., 163., 146., 190., 181., 192., 176.,
       172., 148., 165., 196., 161., 187., 212., 170., 203., 157., 185.,
       130., 174., 207., 134., 141., 152., 179., 132., 201., 198., 209.,
       214., 143., 205., 137., 194., 216., 139., 220., 126., 218., 123.,
       227., 128., 223., 225., 121., 115., 117., 236., 229., 243.,  nan,
       110., 119., 234.])

In [80]:
fifa.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18207 entries, 0 to 18206
Data columns (total 89 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                18207 non-null  int64  
 1   ID                        18207 non-null  int64  
 2   Name                      18207 non-null  object 
 3   Age                       18207 non-null  int64  
 4   Photo                     18207 non-null  object 
 5   Nationality               18207 non-null  object 
 6   Flag                      18207 non-null  object 
 7   Overall                   18207 non-null  int64  
 8   Potential                 18207 non-null  int64  
 9   Club                      17966 non-null  object 
 10  Club Logo                 18207 non-null  object 
 11  Value                     18207 non-null  object 
 12  Wage                      18207 non-null  object 
 13  Special                   18207 non-null  int64  
 14  Prefer

## To find the Unique values of all the columns  

In [81]:
for i in fifa.columns:
    print(i,"------------",fifa[i].unique())

Unnamed: 0 ------------ [    0     1     2 ... 18204 18205 18206]
ID ------------ [158023  20801 190871 ... 241638 246268 246269]
Name ------------ ['L. Messi' 'Cristiano Ronaldo' 'Neymar Jr' ... 'B. Worman'
 'D. Walker-Rice' 'G. Nugent']
Age ------------ [31 33 26 27 32 25 29 28 24 30 19 40 22 23 34 35 36 37 21 18 20 39 41 17
 38 45 42 16 44]
Photo ------------ ['https://cdn.sofifa.org/players/4/19/158023.png'
 'https://cdn.sofifa.org/players/4/19/20801.png'
 'https://cdn.sofifa.org/players/4/19/190871.png' ...
 'https://cdn.sofifa.org/players/4/19/241638.png'
 'https://cdn.sofifa.org/players/4/19/246268.png'
 'https://cdn.sofifa.org/players/4/19/246269.png']
Nationality ------------ ['Argentina' 'Portugal' 'Brazil' 'Spain' 'Belgium' 'Croatia' 'Uruguay'
 'Slovenia' 'Poland' 'Germany' 'France' 'England' 'Italy' 'Egypt'
 'Colombia' 'Denmark' 'Gabon' 'Wales' 'Senegal' 'Costa Rica' 'Slovakia'
 'Netherlands' 'Bosnia Herzegovina' 'Morocco' 'Serbia' 'Algeria' 'Austria'
 'Greece' 'Chile' 'Swe

## To find the missing values

In [82]:
fifa.isnull().sum()

Unnamed: 0           0
ID                   0
Name                 0
Age                  0
Photo                0
                  ... 
GKHandling          48
GKKicking           48
GKPositioning       48
GKReflexes          48
Release Clause    1564
Length: 89, dtype: int64

## To find the missing values %

In [83]:
for i in fifa.columns:
    if fifa[i].isnull().sum()>0:
        print(i,"----------------",fifa[i].isnull().sum()*100/fifa.shape[0],"%")

Club ---------------- 1.323666721590597 %
Preferred Foot ---------------- 0.26363486571099026 %
International Reputation ---------------- 0.26363486571099026 %
Weak Foot ---------------- 0.26363486571099026 %
Skill Moves ---------------- 0.26363486571099026 %
Work Rate ---------------- 0.26363486571099026 %
Body Type ---------------- 0.26363486571099026 %
Real Face ---------------- 0.26363486571099026 %
Position ---------------- 0.32954358213873786 %
Jersey Number ---------------- 0.32954358213873786 %
Joined ---------------- 8.529686384357664 %
Loaned From ---------------- 93.05761520294392 %
Contract Valid Until ---------------- 1.5873015873015872 %
Height ---------------- 0.26363486571099026 %
Weight ---------------- 0.26363486571099026 %
LS ---------------- 11.45163947932114 %
ST ---------------- 11.45163947932114 %
RS ---------------- 11.45163947932114 %
LW ---------------- 11.45163947932114 %
LF ---------------- 11.45163947932114 %
CF ---------------- 11.45163947932114 %
RF -----

In [94]:
fifa.drop(columns=["Loaned From"],inplace=True)

In [96]:
fifa.drop(columns=["Unnamed: 0"],inplace=True)

In [97]:
fifa.shape

(18207, 87)

In [98]:
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  Composure Marking  StandingTackle SlidingTackle  GKDiving  GKHandling  \
0      96.0    33.0            28.0          26.0       6.0        11.0   
1      95.0    28.0            31.0          23.0       7.0        11.0   
2      94.0    27.0            24.0          33.0       9.0         9.0   
3      68.0    15.0            21.0          13.0      90.0        85.0   
4      88.0    68.0            58.0          51.0      15.0        13.0   

   GKKicking GKPositioning GKReflexes Release Clause  
0       15.0          14.0        8.0        €226.5M  
1       15.0          14.0       11.0        €127.1M  
2       15.0          15.0       11.0        €228.1M  
3       87.0          88.0       94.0        €138.6M  
4        5.0          10.0       13.0        €196.4M  

[5 rows x 87 columns]

## Imputation the missing values

In [103]:
for i in fifa .columns:
    if fifa[i].isnull().any():
        if fifa[i].dtypes=="object":
            fifa[i].fillna(fifa[i].mode()[0],inplace=True)
        elif  fifa[i].dtypes=="float":
            fifa[i].fillna(fifa[i].median(),inplace=True)
        elif  fifa[i].dtypes=="int":
            fifa[i].fillna(fifa[i].median(),inplace=True)
        else:
            pass

In [105]:
fifa.isnull().sum().sum()

0

## Photo

In [106]:
fifa["Photo"].str.split("/",expand=True)[6]

0        158023.png
1         20801.png
2        190871.png
3        193080.png
4        192985.png
            ...    
18202    238813.png
18203    243165.png
18204    241638.png
18205    246268.png
18206    246269.png
Name: 6, Length: 18207, dtype: object

In [108]:
fifa["Photo_number"]=fifa["Photo"].str.split("/",expand=True)[6].str.split(".",expand=True)[0]
fifa

ID                Name  Age  \
0      158023            L. Messi   31   
1       20801   Cristiano Ronaldo   33   
2      190871           Neymar Jr   26   
3      193080              De Gea   27   
4      192985        K. De Bruyne   27   
...       ...                 ...  ...   
18202  238813        J. Lundstram   19   
18203  243165  N. Christoffersson   19   
18204  241638           B. Worman   16   
18205  246268      D. Walker-Rice   17   
18206  246269           G. Nugent   16   

                                                Photo Nationality  \
0      https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1       https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2      https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3      https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4      https://cdn.sofifa.org/players/4/19/192985.png     Belgium   
...                                               ...         ...   
18202  https://cdn.sofifa.org/players/4/19/238813.png     England   
18203  https://cdn.sofifa.org/players/4/19/243165.png      Sweden   
18204  https://cdn.sofifa.org/players/4/19/241638.png     England   
18205  https://cdn.sofifa.org/players/4/19/246268.png     England   
18206  https://cdn.sofifa.org/players/4/19/246269.png     England   

                                      Flag  Overall  Potential  \
0      https://cdn.sofifa.org/flags/52.png       94         94   
1      https://cdn.sofifa.org/flags/38.png       94         94   
2      https://cdn.sofifa.org/flags/54.png       92         93   
3      https://cdn.sofifa.org/flags/45.png       91         93   
4       https://cdn.sofifa.org/flags/7.png       91         92   
...                                    ...      ...        ...   
18202  https://cdn.sofifa.org/flags/14.png       47         65   
18203  https://cdn.sofifa.org/flags/46.png       47         63   
18204  https://cdn.sofifa.org/flags/14.png       47         67   
18205  https://cdn.sofifa.org/flags/14.png       47         66   
18206  https://cdn.sofifa.org/flags/14.png       46         66   

                      Club                                       Club Logo  \
0             FC Barcelona    https://cdn.sofifa.org/teams/2/light/241.png   
1                 Juventus     https://cdn.sofifa.org/teams/2/light/45.png   
2      Paris Saint-Germain     https://cdn.sofifa.org/teams/2/light/73.png   
3        Manchester United     https://cdn.sofifa.org/teams/2/light/11.png   
4          Manchester City     https://cdn.sofifa.org/teams/2/light/10.png   
...                    ...                                             ...   
18202      Crewe Alexandra    https://cdn.sofifa.org/teams/2/light/121.png   
18203       Trelleborgs FF    https://cdn.sofifa.org/teams/2/light/703.png   
18204     Cambridge United   https://cdn.sofifa.org/teams/2/light/1944.png   
18205      Tranmere Rovers  https://cdn.sofifa.org/teams/2/light/15048.png   
18206      Tranmere Rovers  https://cdn.sofifa.org/teams/2/light/15048.png   

       ... Marking StandingTackle  SlidingTackle GKDiving  GKHandling  \
0      ...    33.0           28.0           26.0      6.0        11.0   
1      ...    28.0           31.0           23.0      7.0        11.0   
2      ...    27.0           24.0           33.0      9.0         9.0   
3      ...    15.0           21.0           13.0     90.0        85.0   
4      ...    68.0           58.0           51.0     15.0        13.0   
...    ...     ...            ...            ...      ...         ...   
18202  ...    40.0           48.0           47.0     10.0        13.0   
18203  ...    22.0           15.0           19.0     10.0         9.0   
18204  ...    32.0           13.0           11.0      6.0         5.0   
18205  ...    20.0           25.0           27.0     14.0         6.0   
18206  ...    40.0           43.0           50.0     10.0        15.0   

       GKKicking  GKPositioning GKReflexes Release Clause Photo_number

## Flag

In [111]:
fifa["Flag"].str.split("/",expand=True)[4]

0        52.png
1        38.png
2        54.png
3        45.png
4         7.png
          ...  
18202    14.png
18203    46.png
18204    14.png
18205    14.png
18206    14.png
Name: 4, Length: 18207, dtype: object

In [112]:
fifa["Flag_number"]=fifa["Flag"].str.split("/",expand=True)[4].str.split(".",expand=True)[0]
fifa

ID                Name  Age  \
0      158023            L. Messi   31   
1       20801   Cristiano Ronaldo   33   
2      190871           Neymar Jr   26   
3      193080              De Gea   27   
4      192985        K. De Bruyne   27   
...       ...                 ...  ...   
18202  238813        J. Lundstram   19   
18203  243165  N. Christoffersson   19   
18204  241638           B. Worman   16   
18205  246268      D. Walker-Rice   17   
18206  246269           G. Nugent   16   

                                                Photo Nationality  \
0      https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1       https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2      https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3      https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4      https://cdn.sofifa.org/players/4/19/192985.png     Belgium   
...                                               ...         ...   
18202  https://cdn.sofifa.org/players/4/19/238813.png     England   
18203  https://cdn.sofifa.org/players/4/19/243165.png      Sweden   
18204  https://cdn.sofifa.org/players/4/19/241638.png     England   
18205  https://cdn.sofifa.org/players/4/19/246268.png     England   
18206  https://cdn.sofifa.org/players/4/19/246269.png     England   

                                      Flag  Overall  Potential  \
0      https://cdn.sofifa.org/flags/52.png       94         94   
1      https://cdn.sofifa.org/flags/38.png       94         94   
2      https://cdn.sofifa.org/flags/54.png       92         93   
3      https://cdn.sofifa.org/flags/45.png       91         93   
4       https://cdn.sofifa.org/flags/7.png       91         92   
...                                    ...      ...        ...   
18202  https://cdn.sofifa.org/flags/14.png       47         65   
18203  https://cdn.sofifa.org/flags/46.png       47         63   
18204  https://cdn.sofifa.org/flags/14.png       47         67   
18205  https://cdn.sofifa.org/flags/14.png       47         66   
18206  https://cdn.sofifa.org/flags/14.png       46         66   

                      Club                                       Club Logo  \
0             FC Barcelona    https://cdn.sofifa.org/teams/2/light/241.png   
1                 Juventus     https://cdn.sofifa.org/teams/2/light/45.png   
2      Paris Saint-Germain     https://cdn.sofifa.org/teams/2/light/73.png   
3        Manchester United     https://cdn.sofifa.org/teams/2/light/11.png   
4          Manchester City     https://cdn.sofifa.org/teams/2/light/10.png   
...                    ...                                             ...   
18202      Crewe Alexandra    https://cdn.sofifa.org/teams/2/light/121.png   
18203       Trelleborgs FF    https://cdn.sofifa.org/teams/2/light/703.png   
18204     Cambridge United   https://cdn.sofifa.org/teams/2/light/1944.png   
18205      Tranmere Rovers  https://cdn.sofifa.org/teams/2/light/15048.png   
18206      Tranmere Rovers  https://cdn.sofifa.org/teams/2/light/15048.png   

       ... StandingTackle SlidingTackle  GKDiving GKHandling  GKKicking  \
0      ...           28.0          26.0       6.0       11.0       15.0   
1      ...           31.0          23.0       7.0       11.0       15.0   
2      ...           24.0          33.0       9.0        9.0       15.0   
3      ...           21.0          13.0      90.0       85.0       87.0   
4      ...           58.0          51.0      15.0       13.0        5.0   
...    ...            ...           ...       ...        ...        ...   
18202  ...           48.0          47.0      10.0       13.0        7.0   
18203  ...           15.0          19.0      10.0        9.0        9.0   
18204  ...           13.0          11.0       6.0        5.0       10.0   
18205  ...           25.0          27.0      14.0        6.0       14.0   
18206  ...           43.0          50.0      10.0       15.0        9.0   

       GKPositioning  GKReflexes Release Claus

In [119]:
fifa["LS"].unique()

array(['88+2', '91+3', '84+3', '61+2', '82+3', '83+3', '77+3', '87+5',
       '73+3', '87+3', '78+3', '64+3', '72+3', '86+3', '71+3', '85+3',
       '58+3', '80+3', '76+3', '79+3', '69+3', '66+3', '70+3', '52+3',
       '81+3', '68+3', '82+2', '75+2', '62+3', '74+3', '75+3', '67+3',
       '65+3', '61+3', '57+2', '81+2', '49+3', '74+2', '63+3', '60+3',
       '82+4', '56+2', '64+2', '77+2', '59+3', '70+2', '57+3', '72+2',
       '55+3', '78+2', '59+2', '73+2', '76+2', '52+2', '80+2', '79+2',
       '56+3', '53+2', '58+2', '69+2', '51+3', '66+2', '67+2', '68+2',
       '65+2', '62+2', '55+2', '71+2', '63+2', '60+2', '54+2', '49+2',
       '50+2', '51+2', '48+2', '47+2', '47+3', '46+2', '42+2', '44+2',
       '45+2', '43+2', '40+2', '39+2', '37+2', '41+2', '38+2', '36+2',
       '34+2', '35+2', '31+2', '33+2', '32+2'], dtype=object)

In [114]:
fifa["LS"]

0        88+2
1        91+3
2        84+3
3        61+2
4        82+3
         ... 
18202    42+2
18203    45+2
18204    45+2
18205    47+2
18206    43+2
Name: LS, Length: 18207, dtype: object

In [122]:
fifa["ls"]=fifa["LS"].apply(lambda x:x[0:2])
fifa["ls2"]=fifa["LS"].apply(lambda x:x[-1])
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  GKHandling GKKicking  GKPositioning GKReflexes  Release Clause  \
0       11.0      15.0           14.0        8.0         €226.5M   
1       11.0      15.0           14.0       11.0         €127.1M   
2        9.0      15.0           15.0       11.0         €228.1M   
3       85.0      87.0           88.0       94.0         €138.6M   
4       13.0       5.0           10.0       13.0         €196.4M   

   Photo_number  Flag_number  rm  ls ls2  
0        158023           52  91  88   2  
1         20801           38  88  91   3  
2        190871           54  88  84   3  
3        193080           45  61  61   2  
4        192985            7  88  82   3  

[5 rows x 92 columns]

In [124]:
fifa["ls_sum"]=fifa["LS"].apply(lambda x:eval(x))
fifa["ls_sum"]

0        90
1        94
2        87
3        63
4        85
         ..
18202    44
18203    47
18204    47
18205    49
18206    45
Name: ls_sum, Length: 18207, dtype: int64

In [125]:
fifa["st"]=fifa["ST"].apply(lambda x:x[0:2])
fifa["st2"]=fifa["ST"].apply(lambda x:x[-1])
fifa["st_sum"]=fifa["ST"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  Release Clause Photo_number  Flag_number  rm  ls  ls2  ls_sum  st st2 st_sum  
0        €226.5M       158023           52  91  88    2      90  88   2     90  
1        €127.1M        20801           38  88  91    3      94  91   3     94  
2        €228.1M       190871           54  88  84    3      87  84   3     87  
3        €138.6M       193080           45  61  61    2      63  61   2     63  
4        €196.4M       192985            7  88  82    3      85  82   3     85  

[5 rows x 96 columns]

In [127]:
fifa["rs"]=fifa["RS"].apply(lambda x:x[0:2])
fifa["rs2"]=fifa["RS"].apply(lambda x:x[-1])
fifa["rs_sum"]=fifa["RS"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  rm  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...  91   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...  88   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...  88   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...  61   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...  88   

   ls  ls2 ls_sum  st  st2  st_sum  rs rs2 rs_sum  
0  88    2     90  88    2      90  88   2     90  
1  91    3     94  91    3      94  91   3     94  
2  84    3     87  84    3      87  84   3     87  
3  61    2     63  61    2      63  61   2     63  
4  82    3     85  82    3      85  82   3     85  

[5 rows x 99 columns]

In [128]:
fifa["lw"]=fifa["LW"].apply(lambda x:x[0:2])
fifa["lw2"]=fifa["LW"].apply(lambda x:x[-1])
fifa["lw_sum"]=fifa["LW"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  ls_sum  st  st2 st_sum  rs  rs2  rs_sum  lw lw2 lw_sum  
0     90  88    2     90  88    2      90  92   2     94  
1     94  91    3     94  91    3      94  89   3     92  
2     87  84    3     87  84    3      87  89   3     92  
3     63  61    2     63  61    2      63  63   2     65  
4     85  82    3     85  82    3      85  87   3     90  

[5 rows x 102 columns]

In [129]:
fifa["lf"]=fifa["LF"].apply(lambda x:x[0:2])
fifa["lf2"]=fifa["LF"].apply(lambda x:x[-1])
fifa["lf_sum"]=fifa["LF"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  st_sum  rs  rs2 rs_sum  lw  lw2  lw_sum  lf lf2 lf_sum  
0     90  88    2     90  92    2      94  93   2     95  
1     94  91    3     94  89    3      92  90   3     93  
2     87  84    3     87  89    3      92  89   3     92  
3     63  61    2     63  63    2      65  61   2     63  
4     85  82    3     85  87    3      90  87   3     90  

[5 rows x 105 columns]

In [130]:
fifa["cf"]=fifa["CF"].apply(lambda x:x[0:2])
fifa["cf2"]=fifa["CF"].apply(lambda x:x[-1])
fifa["cf_sum"]=fifa["CF"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  rs_sum  lw  lw2 lw_sum  lf  lf2  lf_sum  cf cf2 cf_sum  
0     90  92    2     94  93    2      95  93   2     95  
1     94  89    3     92  90    3      93  90   3     93  
2     87  89    3     92  89    3      92  89   3     92  
3     63  63    2     65  61    2      63  61   2     63  
4     85  87    3     90  87    3      90  87   3     90  

[5 rows x 108 columns]

In [131]:
fifa["rf"]=fifa["RF"].apply(lambda x:x[0:2])
fifa["rf2"]=fifa["RF"].apply(lambda x:x[-1])
fifa["rf_sum"]=fifa["RF"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  lw_sum  lf  lf2 lf_sum  cf  cf2  cf_sum  rf rf2 rf_sum  
0     94  93    2     95  93    2      95  93   2     95  
1     92  90    3     93  90    3      93  90   3     93  
2     92  89    3     92  89    3      92  89   3     92  
3     65  61    2     63  61    2      63  61   2     63  
4     90  87    3     90  87    3      90  87   3     90  

[5 rows x 111 columns]

In [132]:
fifa["rw"]=fifa["RW"].apply(lambda x:x[0:2])
fifa["rw2"]=fifa["RW"].apply(lambda x:x[-1])
fifa["rw_sum"]=fifa["RW"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  lf_sum  cf  cf2 cf_sum  rf  rf2  rf_sum  rw rw2 rw_sum  
0     95  93    2     95  93    2      95  92   2     94  
1     93  90    3     93  90    3      93  89   3     92  
2     92  89    3     92  89    3      92  89   3     92  
3     63  61    2     63  61    2      63  63   2     65  
4     90  87    3     90  87    3      90  87   3     90  

[5 rows x 114 columns]

In [133]:
fifa["lam"]=fifa["LAM"].apply(lambda x:x[0:2])
fifa["lam2"]=fifa["LAM"].apply(lambda x:x[-1])
fifa["lam_sum"]=fifa["LAM"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  cf_sum  rf  rf2 rf_sum  rw  rw2  rw_sum lam lam2 lam_sum  
0     95  93    2     95  92    2      94  93    2      95  
1     93  90    3     93  89    3      92  88    3      91  
2     92  89    3     92  89    3      92  89    3      92  
3     63  61    2     63  63    2      65  61    2      63  
4     90  87    3     90  87    3      90  88    3      91  

[5 rows x 117 columns]

In [134]:
fifa["cam"]=fifa["CAM"].apply(lambda x:x[0:2])
fifa["cam2"]=fifa["CAM"].apply(lambda x:x[-1])
fifa["cam_sum"]=fifa["CAM"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  rf_sum  rw  rw2 rw_sum  lam  lam2  lam_sum cam cam2 cam_sum  
0     95  92    2     94   93     2       95  93    2      95  
1     93  89    3     92   88     3       91  88    3      91  
2     92  89    3     92   89     3       92  89    3      92  
3     63  63    2     65   61     2       63  61    2      63  
4     90  87    3     90   88     3       91  88    3      91  

[5 rows x 120 columns]

In [135]:
fifa["ram"]=fifa["RAM"].apply(lambda x:x[0:2])
fifa["ram2"]=fifa["RAM"].apply(lambda x:x[-1])
fifa["ram_sum"]=fifa["RAM"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  rw_sum lam  lam2 lam_sum  cam  cam2  cam_sum ram ram2 ram_sum  
0     94  93     2      95   93     2       95  93    2      95  
1     92  88     3      91   88     3       91  88    3      91  
2     92  89     3      92   89     3       92  89    3      92  
3     65  61     2      63   61     2       63  61    2      63  
4     90  88     3      91   88     3       91  88    3      91  

[5 rows x 123 columns]

In [136]:
fifa["lm"]=fifa["LM"].apply(lambda x:x[0:2])
fifa["lm2"]=fifa["LM"].apply(lambda x:x[-1])
fifa["lm_sum"]=fifa["LM"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  lam_sum cam  cam2 cam_sum  ram  ram2  ram_sum  lm lm2 lm_sum  
0      95  93     2      95   93     2       95  91   2     93  
1      91  88     3      91   88     3       91  88   3     91  
2      92  89     3      92   89     3       92  88   3     91  
3      63  61     2      63   61     2       63  61   2     63  
4      91  88     3      91   88     3       91  88   3     91  

[5 rows x 126 columns]

In [137]:
fifa["lcm"]=fifa["LCM"].apply(lambda x:x[0:2])
fifa["lcm2"]=fifa["LCM"].apply(lambda x:x[-1])
fifa["lcm_sum"]=fifa["LCM"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  cam_sum ram  ram2 ram_sum  lm  lm2  lm_sum lcm lcm2 lcm_sum  
0      95  93     2      95  91    2      93  84    2      86  
1      91  88     3      91  88    3      91  81    3      84  
2      92  89     3      92  88    3      91  81    3      84  
3      63  61     2      63  61    2      63  58    2      60  
4      91  88     3      91  88    3      91  87    3      90  

[5 rows x 129 columns]

In [138]:
fifa["cm"]=fifa["CM"].apply(lambda x:x[0:2])
fifa["cm2"]=fifa["CM"].apply(lambda x:x[-1])
fifa["cm_sum"]=fifa["CM"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  ram_sum  lm  lm2 lm_sum  lcm  lcm2  lcm_sum  cm cm2 cm_sum  
0      95  91    2     93   84     2       86  84   2     86  
1      91  88    3     91   81     3       84  81   3     84  
2      92  88    3     91   81     3       84  81   3     84  
3      63  61    2     63   58     2       60  58   2     60  
4      91  88    3     91   87     3       90  87   3     90  

[5 rows x 132 columns]

In [139]:
fifa["rcm"]=fifa["RCM"].apply(lambda x:x[0:2])
fifa["rcm2"]=fifa["RCM"].apply(lambda x:x[-1])
fifa["rcm_sum"]=fifa["RCM"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  lm_sum lcm  lcm2 lcm_sum  cm  cm2  cm_sum rcm rcm2 rcm_sum  
0     93  84     2      86  84    2      86  84    2      86  
1     91  81     3      84  81    3      84  81    3      84  
2     91  81     3      84  81    3      84  81    3      84  
3     63  58     2      60  58    2      60  58    2      60  
4     91  87     3      90  87    3      90  87    3      90  

[5 rows x 135 columns]

In [140]:
fifa["rm"]=fifa["RM"].apply(lambda x:x[0:2])
fifa["rm2"]=fifa["RM"].apply(lambda x:x[-1])
fifa["rm_sum"]=fifa["RM"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  lcm2 lcm_sum  cm cm2  cm_sum  rcm  rcm2 rcm_sum rm2 rm_sum  
0    2      86  84   2      86   84     2      86   2     93  
1    3      84  81   3      84   81     3      84   3     91  
2    3      84  81   3      84   81     3      84   3     91  
3    2      60  58   2      60   58     2      60   2     63  
4    3      90  87   3      90   87     3      90   3     91  

[5 rows x 137 columns]

In [141]:
fifa["lwb"]=fifa["LWB"].apply(lambda x:x[0:2])
fifa["lwb2"]=fifa["LWB"].apply(lambda x:x[-1])
fifa["lwb_sum"]=fifa["LWB"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ... cm2  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   2   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   3   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   3   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   2   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   3   

  cm_sum  rcm rcm2  rcm_sum  rm2  rm_sum lwb lwb2 lwb_sum  
0     86   84    2       86    2      93  64    2      66  
1     84   81    3       84    3      91  65    3      68  
2     84   81    3       84    3      91  65    3      68  
3     60   58    2       60    2      63  61    2      63  
4     90   87    3       90    3      91  77    3      80  

[5 rows x 140 columns]

In [142]:
fifa["ldm"]=fifa["LDM"].apply(lambda x:x[0:2])
fifa["ldm2"]=fifa["LDM"].apply(lambda x:x[-1])
fifa["ldm_sum"]=fifa["LDM"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  rcm2 rcm_sum  rm2 rm_sum  lwb  lwb2  lwb_sum ldm ldm2 ldm_sum  
0    2      86    2     93   64     2       66  61    2      63  
1    3      84    3     91   65     3       68  61    3      64  
2    3      84    3     91   65     3       68  60    3      63  
3    2      60    2     63   61     2       63  62    2      64  
4    3      90    3     91   77     3       80  77    3      80  

[5 rows x 143 columns]

In [143]:
fifa["cdm"]=fifa["CDM"].apply(lambda x:x[0:2])
fifa["cdm2"]=fifa["CDM"].apply(lambda x:x[-1])
fifa["cdm_sum"]=fifa["CDM"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  rm_sum lwb  lwb2 lwb_sum  ldm  ldm2  ldm_sum cdm cdm2 cdm_sum  
0     93  64     2      66   61     2       63  61    2      63  
1     91  65     3      68   61     3       64  61    3      64  
2     91  65     3      68   60     3       63  60    3      63  
3     63  61     2      63   62     2       64  62    2      64  
4     91  77     3      80   77     3       80  77    3      80  

[5 rows x 146 columns]

In [144]:
fifa["rdm"]=fifa["RDM"].apply(lambda x:x[0:2])
fifa["rdm2"]=fifa["RDM"].apply(lambda x:x[-1])
fifa["rdm_sum"]=fifa["RDM"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  lwb_sum ldm  ldm2 ldm_sum  cdm  cdm2  cdm_sum rdm rdm2 rdm_sum  
0      66  61     2      63   61     2       63  61    2      63  
1      68  61     3      64   61     3       64  61    3      64  
2      68  60     3      63   60     3       63  60    3      63  
3      63  62     2      64   62     2       64  62    2      64  
4      80  77     3      80   77     3       80  77    3      80  

[5 rows x 149 columns]

In [145]:
fifa["rwb"]=fifa["RWB"].apply(lambda x:x[0:2])
fifa["rwb2"]=fifa["RWB"].apply(lambda x:x[-1])
fifa["rwb_sum"]=fifa["RWB"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  ldm_sum cdm  cdm2 cdm_sum  rdm  rdm2  rdm_sum rwb rwb2 rwb_sum  
0      63  61     2      63   61     2       63  64    2      66  
1      64  61     3      64   61     3       64  65    3      68  
2      63  60     3      63   60     3       63  65    3      68  
3      64  62     2      64   62     2       64  61    2      63  
4      80  77     3      80   77     3       80  77    3      80  

[5 rows x 152 columns]

In [146]:
fifa["lb"]=fifa["LB"].apply(lambda x:x[0:2])
fifa["lb2"]=fifa["LB"].apply(lambda x:x[-1])
fifa["lb_sum"]=fifa["LB"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  cdm_sum rdm  rdm2 rdm_sum  rwb  rwb2  rwb_sum  lb lb2 lb_sum  
0      63  61     2      63   64     2       66  59   2     61  
1      64  61     3      64   65     3       68  61   3     64  
2      63  60     3      63   65     3       68  60   3     63  
3      64  62     2      64   61     2       63  61   2     63  
4      80  77     3      80   77     3       80  73   3     76  

[5 rows x 155 columns]

In [147]:
fifa["lcb"]=fifa["LCB"].apply(lambda x:x[0:2])
fifa["lcb2"]=fifa["LCB"].apply(lambda x:x[-1])
fifa["lcb_sum"]=fifa["LCB"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  rdm_sum rwb  rwb2 rwb_sum  lb  lb2  lb_sum lcb lcb2 lcb_sum  
0      63  64     2      66  59    2      61  47    2      49  
1      64  65     3      68  61    3      64  53    3      56  
2      63  65     3      68  60    3      63  47    3      50  
3      64  61     2      63  61    2      63  63    2      65  
4      80  77     3      80  73    3      76  66    3      69  

[5 rows x 158 columns]

CDM                       18207 non-null  object 
 45  RDM                       18207 non-null  object 
 46  RWB                       18207 non-null  object 
 47  LB                        18207 non-null  object 
 48  LCB                       18207 non-null  object 
 49  CB                        18207 non-null  object 
 50  RCB                       18207 non-null  object 
 51  RB   

In [148]:
fifa["cb"]=fifa["CB"].apply(lambda x:x[0:2])
fifa["cb2"]=fifa["CB"].apply(lambda x:x[-1])
fifa["cb_sum"]=fifa["CB"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  rwb_sum  lb  lb2 lb_sum  lcb  lcb2  lcb_sum  cb cb2 cb_sum  
0      66  59    2     61   47     2       49  47   2     49  
1      68  61    3     64   53     3       56  53   3     56  
2      68  60    3     63   47     3       50  47   3     50  
3      63  61    2     63   63     2       65  63   2     65  
4      80  73    3     76   66     3       69  66   3     69  

[5 rows x 161 columns]

In [149]:
fifa["rcb"]=fifa["RCB"].apply(lambda x:x[0:2])
fifa["rcb2"]=fifa["RCB"].apply(lambda x:x[-1])
fifa["rcb_sum"]=fifa["RCB"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  lb_sum lcb  lcb2 lcb_sum  cb  cb2  cb_sum rcb rcb2 rcb_sum  
0     61  47     2      49  47    2      49  47    2      49  
1     64  53     3      56  53    3      56  53    3      56  
2     63  47     3      50  47    3      50  47    3      50  
3     63  63     2      65  63    2      65  63    2      65  
4     76  66     3      69  66    3      69  66    3      69  

[5 rows x 164 columns]

In [152]:
fifa["rb"]=fifa["RB"].apply(lambda x:x[0:2])
fifa["rb2"]=fifa["RB"].apply(lambda x:x[-1])
fifa["rb_sum"]=fifa["RB"].apply(lambda x:eval(x))
fifa.head()

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   
2  190871          Neymar Jr   26   
3  193080             De Gea   27   
4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club                                     Club Logo  ...  \
0         FC Barcelona  https://cdn.sofifa.org/teams/2/light/241.png  ...   
1             Juventus   https://cdn.sofifa.org/teams/2/light/45.png  ...   
2  Paris Saint-Germain   https://cdn.sofifa.org/teams/2/light/73.png  ...   
3    Manchester United   https://cdn.sofifa.org/teams/2/light/11.png  ...   
4      Manchester City   https://cdn.sofifa.org/teams/2/light/10.png  ...   

  lcb_sum  cb  cb2 cb_sum  rcb  rcb2  rcb_sum  rb rb2 rb_sum  
0      49  47    2     49   47     2       49  59   2     61  
1      56  53    3     56   53     3       56  61   3     64  
2      50  47    3     50   47     3       50  60   3     63  
3      65  63    2     65   63     2       65  61   2     63  
4      69  66    3     69   66     3       69  73   3     76  

[5 rows x 167 columns]

In [153]:
fifa["Release Clause"].unique()

array(['€226.5M', '€127.1M', '€228.1M', ..., '€74K', '€101K', '€147K'],
      dtype=object)

In [155]:
fifa["Release Clause"]=fifa["Release Clause"].str.replace("€","")

In [157]:
fifa["Release Clause"]=fifa["Release Clause"].str.replace("M","")

In [158]:
fifa["Release Clause"]=fifa["Release Clause"].str.replace("K","")

In [160]:
fifa["Release Clause"].unique()

array(['226.5', '127.1', '228.1', ..., '67', '101', '147'], dtype=object)